streamlit_aggrid_redux/frontend/src/AgGrid.tsx

```diff
@@ -184,12 +175,21 @@ class AgGrid<S = {}> extends React.Component<ComponentProps, S> {
                 LicenseManager.setLicenseKey(props.args["license_key"])
             }
             else {
-                     console.log("Enterprise modules requested without license key; using trial version")
+                console.log("Enterprise modules requested without license key; using trial version")
             }
         }
 
-        this.isGridAutoHeightOn = 
-            this.props.args.grid_options?.domLayout === "autoHeight"
+        // add some items from input grid options
+        this.isGridAutoHeightOn = this.props.args.grid_options?.domLayout === "autoHeight"
+
+        if (("clearCheckboxOnReload" in this.props.args.grid_options)) {
+            this.clearCheckboxOnReload = this.props.args.grid_options["clearCheckboxOnReload"];
+            delete this.props.args.grid_options["clearCheckboxOnReload"];
+        }
+        if (("preSelectAllRows" in this.props.args.grid_options)) {
+            this.preSelectAllRows = this.props.args.grid_options["preSelectAllRows"];
+            delete this.props.args.grid_options["preSelectAllRows"];
+        }
```

Comment:
I was getting JS warnings about invalid grid_options parameters, so I check and remove these